src/client/js/app.js

```diff
@@ -0,0 +1,29 @@
+/**
```

Comment:
If I understand webpack correctly, this compiles into main.js, the main entrypoint in webpack config. This suggests it's a main file for the whole application. But in reality this is used (and useful) only for the "/listing" page. I'd either change the name of the file, or it's content (e.g. empty for now? since it's actually a placeholder for concatenated scripts?), because otherwise I'm worried we'll end up initializing firebase and pulling products for every page unnecessarily.

Reply:
Yes, sorry I wasn't very clear in my commit comment. This was designed as a stop-gap measure, with the end goal being per-route bundling, as you suggest. I did things this way for now because `/listing` is really the only route with JS so far, but it sounds like you think it's worth it to invest some extra time now and do per-route bundling from the beginning?

Reply:
I think we should have a strategy for loading in general. Right now we serve individual files - which is great for http2 (yet they should get preprocessed, minified etc). This PR changes it to make one big bundle for whole app served on every subpage. I have a feeling that in no time we'll be rewriting it back to have more granular scripts back. So maybe its not worth rewriting bundling back and forth, but come up with a solution that we intend to develop in the long run? Otherwise we'll spend a lot of time rewiring things. 

Since we do not intend to stay with the big bundle, maybe it's better to leave individual files in dev mode for now? And bundle them only once we know how exactly they should be structured?

I think the final structure will be more complex than just bundle-per-subpage, especially if we want to do app shell model.

Reply:
Unfortunately, if we leave things as individual files in dev, things will really only work with Chrome and Safari, since they're the only two browsers supporting native imports at the moment, and even then we might need to enable flags once we start doing dynamic imports.

It's also probably a good idea to be able to test out our bundling strategy in dev, where we have sourcemaps and the like available.

I'll put some work into this now and try to come up with a future-proof strategy that takes into account everything we want to be able to do.

Reply:
So in other words, we need webpack bundling right now because we want to use ES6 imports, right? If so, let's check it in as is (maybe optionally rename main.js, since it's useful only for /listing) to unlock folks that want to add more js code (as ideally we want them to write in es6). Then let's come up with final loading strategy in a next PR. Wdyt?

Reply:
Yes, sounds about right. As an interesting side benefit, it will also give us some authority on how to break up a single-bundle application into multiple pieces, something that we advocate to developers a lot. If we go through the pain ourselves, even if it's with a small set of files to start with, it should give us some insight into what people have to go through.

I'll leave it as `main.js`, to really drive home the fact that all routes will end up there and some sort of handling will be needed so that the right code gets executed for each route. The next PR would then add in `/home` and split everything nicely.